engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/validator/LiteralValidatorExtractor.scala

```diff
@@ -2,15 +2,14 @@ package pl.touk.nussknacker.engine.definition.validator
 
 import java.lang.reflect.Parameter
 
-import pl.touk.nussknacker.engine.api.definition.{LiteralIntValidator, ParameterValidator}
+import pl.touk.nussknacker.engine.api.definition.{LiteralParameterValidator, ParameterValidator}
 import pl.touk.nussknacker.engine.api.validation.Literal
 
 object LiteralValidatorExtractor extends ValidatorExtractor {
   override def extract(param: Parameter): Option[ParameterValidator] =
     param.getAnnotation(classOf[Literal]) match {
-      case _:Literal => param.getType match {
-        case clazz if clazz == classOf[Int] => Some(LiteralIntValidator)
-        case clazz if clazz == classOf[Integer] => Some(LiteralIntValidator)
+      case _: Literal => param.getType match {
+        case clazz => LiteralParameterValidator(clazz)
```

Comment:
I think this match is futile? You can just write LiteralParameterValidator(param.getType)?
And what about Option/Optional/LazyParameter? Will it work for them (I know it's kind of stupid to require Literal on LazyParameter or Option, but still - maybe it deserves at least some comment?)

Reply:
It's not true.. When we have Integer param with RAW editr then we can put there expresion like: `#input.value` and more.. In that case LiteralValidator throws error.. So I think better way is use annotation when we want use Literal.. 

Ok, we don't support Option/Optional - these types are not Literal.. But we support `@Nullable` :)  I left some comment.